fineract-core/src/main/java/org/apache/fineract/portfolio/savings/data/SavingsAccountTransactionData.java

```diff
@@ -129,7 +129,7 @@ public static SavingsAccountTransactionData overdraftInterest(final SavingsAccou
                 savingsAccountTransactionType.getValue().longValue(), savingsAccountTransactionType.getCode(),
                 savingsAccountTransactionType.getValue().toString());
         return new SavingsAccountTransactionData(amount.getAmount(), date, savingsAccount.getId(), savingsAccountTransactionEnumData,
-                isReversed, null, isManualTransaction, lienTransaction);
+                isReversed, null, isManualTransaction, lienTransaction, date);
```

Comment:
This looks wrong to me. We should not set the submitted on date by the transaction date.

Reply:
Same interest posting period